### PR TITLE
execute tests on junit platform

### PR DESCRIPTION
**Pull request by epeee**
a very first step towards #1144

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -91,6 +97,40 @@
...
+      <scope>test</scope>
+    </dependency>
+    <!-- Only required to run tests in an IDE that bundles an older version -->
+    <dependency>
+      <groupId>org.junit.vintage</groupId>
+      <artifactId>junit-vintage-engine</artifactId>
+      <version>${junit.vintage.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <!-- To avoid compiler warnings about @API annotations in JUnit code -->
```
Comment (filiphr) [inline pom.xml]: I think that they returned the transitive dependency on apiguardian with [5.0.3](http://junit.org/junit5/docs/current/user-guide/#release-notes-5.0.3-overall-improvements)

Comment (filiphr) [reply pom.xml]: Just adding that part as piece of information, not that it is something crucial. I am not member of the project, but I am excited for adding JUnit 5 :)

Comment (epeee) [reply pom.xml]: @filiphr thx for pointing that out!
I will update the pr accordingly.

**Comment by joel-costigliola** [comment]
I believe this has been addressed in https://github.com/joel-costigliola/assertj-core/commit/393b9f36529746d646d6d5629354b5cb0455ad9e

**Pull request closed without merging**